src/seeds.scheduler.cpp

```diff
@@ -93,29 +165,50 @@ ACTION scheduler::execute() {
     */
 
     auto itr = operations.begin();
+
+    auto it_s = config.find(seconds_to_execute.value);
+    check(it_s != config.end(), contracts::scheduler.to_string() + ": the parameter " + seconds_to_execute.to_string() + " is not configured in " + contracts::settings.to_string());
+
+    action next_execution(
+        permission_level{get_self(), "active"_n},
+        get_self(),
+        "execute"_n,
+        std::make_tuple()
+    );
+
+    print("Creating the diferred transaction...");
+
+    transaction tx;
+    tx.actions.emplace_back(next_execution);
+    tx.delay_sec = it_s -> value;
+    tx.send(contracts::scheduler.value /*eosio::current_time_point().sec_since_epoch() + 30*/, _self);
+    
+    print("transaction sended");
+
     while(itr != operations.end()) {
-        if(isRdyToExec(itr -> operation)){
-            
+        if(isRdyToExec(itr -> id)){
+
+            print("Operation to be executed: " + itr -> id.to_string());
+
             action a = action(
-                //permission_level{contracts::forum, "period"_n},
-                //permission_level(get_self(), "scheduled"_n),
-                permission_level{get_self(), "active"_n},
+                permission_level{itr -> contract, "execute"_n},
                 itr -> contract,
                 itr -> operation,
                 std::make_tuple()
             );
 
-            a.send();
+            transaction txa;
```

Comment:
@IanMendozaJaimes @7flash So this is sending a delayed transaction for each scheduler call with a delay of 0?

I also see that every time this executes, it will only execute 1 scheduled operation. That's good because then we have enough CPU. But I think we need to then iterate over it sorted by last execution date (timestamp) so we make sure we never forget any operations - if we always go by the op name, then it could be that we always execute the first op, but the other ops never even get looked at.

I'll make this change